walk/walk.go

```diff
@@ -333,5 +332,5 @@ func resolveFileInfo(wc *walkConfig, dir, rel string, fi fs.FileInfo) fs.FileInf
 		// A symlink, but not one we could resolve.
 		return nil
 	}
-	return fi
+	return fs.FileInfoToDirEntry(fi)
```

Comment:
Are we actually using the extra metadata?

Reply:
No, I think we `stat` here to resolve the symlink. We could use `os.Readlink` instead but then we won't know if we landed on a file or a dir, so it seems we'd need a `stat` either way. But also symlinks in source repos should be rare so I'm not worried about trying to optimize this path.